### DISCUSSION TITLE
RavenDB Studio - Related documents not showing

### DISCUSSION THREAD

**Discussion by aventus13** [ravendb/ravendb#12137] [Q&A] [answered]
I'm trying to understand how does RavenDB handle document relations. As depicted on the screenshot below, I have a collection called `Expenses`. Each expense contains a property `UserId` which points at a user, stored in the `Users` collection. On this example, my expense document contains a `UserId` of an existing, valid document in `Users` collection. Yet the studio interface doesn't seem to see this relation. Am I missing something?

![image](https://user-images.githubusercontent.com/42281325/117848228-00dae880-b27b-11eb-8751-5ed7782337a2.png)

**Comment by garayx** [comment, marked as answer]
This view is based on the id format. 
Guid type Ids won't be shown in the related documents list.
![image](https://user-images.githubusercontent.com/16625387/117853484-27a41980-b291-11eb-9d71-069780e49605.png)

I have opened an issue to fix it in the future, [https://issues.hibernatingrhinos.com/issue/RavenDB-16665](https://issues.hibernatingrhinos.com/issue/RavenDB-16665)

**Comment by aventus13** [comment]
Thanks for explaining this and opening an issue @garayx. I don't know the inner workings of Studio's related documents feature, but purely from a user's perspective it makes sense to relate any document irrelevant of the ID format, given that document IDs are unique across the entire database in RavenDB and IDs can be just arbitrary strings.

**Reply by ppekrol** [reply]
We will look into GUIDs, or at least some of the known formats, but this is a complex problem. By definition RavenDB is schema-less database, so we do not know if a string field actually represents an ID or not. Checking each string would be a very expensive operation.
